src/components/themed/FlipInput.js

```diff
@@ -0,0 +1,570 @@
+            if (this.state.isToggled) {
+              this.onToggleFlipInput()
+            }
+          } else {
+            this.textInputBack && this.textInputBack.focus()
+          }
+        } else {
+          this.textInputFront && this.textInputFront.focus()
+        }
+      }, 650)
+    }
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps: Props) {
+    // Check if primary changed first. Don't bother to check secondary if parent passed in a primary
+    if (
+      nextProps.overridePrimaryDecimalAmount !== this.state.overridePrimaryDecimalAmount ||
+      nextProps.forceUpdateGuiCounter !== this.state.forceUpdateGuiCounter
+    ) {
+      const result = setPrimaryToSecondary(nextProps, sanitizeDecimalAmount(nextProps.overridePrimaryDecimalAmount, nextProps.primaryInfo.maxEntryDecimals))
+      this.setState({
+        ...result,
+        overridePrimaryDecimalAmount: nextProps.overridePrimaryDecimalAmount,
+        forceUpdateGuiCounter: nextProps.forceUpdateGuiCounter
+      })
+    } else {
+      // Checks and apply exchange rates
+      if (!this.state.isToggled) {
+        this.setState(setPrimaryToSecondary(nextProps, this.state.primaryDecimalAmount))
+      } else {
+        this.setState(setSecondaryToPrimary(nextProps, this.state.secondaryDecimalAmount))
+      }
+    }
+    if (nextProps.primaryInfo.currencyCode !== this.props.primaryInfo.currencyCode) {
+      setTimeout(() => this.onKeyPress('0', '', this.props.primaryInfo.maxEntryDecimals, setPrimaryToSecondary), 50)
+    }
+  }
+
+  onToggleFlipInput = () => {
+    this.clipboardMenu.close()
+    this.setState({
+      isToggled: !this.state.isToggled
+    })
+    if (this.state.isToggled) {
+      if (this.textInputFront) {
+        this.textInputFront.focus()
+      }
+      Animated.spring(this.animatedValue, {
+        toValue: 0,
+        friction: 8,
+        tension: 10
+      }).start()
+    }
+    if (!this.state.isToggled) {
+      if (this.textInputBack) {
+        this.textInputBack.focus()
+      }
+      Animated.spring(this.animatedValue, {
+        toValue: 1,
+        friction: 8,
+        tension: 10
+      }).start()
+    }
+  }
+
+  async openClipboardMenu() {
+    this.clipboardMenu.close()
+    try {
+      if ((this.state.textInputFrontFocus || this.state.textInputBackFocus) && (await Clipboard.getString())) {
+        this.clipboardMenu.open()
+      }
+    } catch (error) {
+      showError(error)
+    }
+  }
+
+  handlePasteClipboard = async () => {
+    try {
+      const clipboard = await Clipboard.getString()
+      if (this.state.isToggled) {
+        this.setStateAmounts('', setSecondaryToPrimary)
+        for (const string of clipboard.split('')) {
+          this.onKeyPress(string, this.state.secondaryDecimalAmount, this.props.secondaryInfo.maxEntryDecimals, setSecondaryToPrimary)
+        }
+      } else {
+        this.setStateAmounts('', setSecondaryToPrimary)
```

Comment:
Shouldn't this be 'setPrimaryToSecondary' and not 'setSecondaryToPrimary'?

Reply:
This is correct because we are resetting the amount, fiat amount is blank while crypto amount is 0